Scripts/LaunchSidechainMasternode.ps1

```diff
@@ -0,0 +1,484 @@
+    Write-Host ""
+}
+
+function Shutdown-SidechainNode
+{
+    Write-Host "Shutting down Sidechain Node..." -ForegroundColor Yellow
+    $Headers = @{}
+    $Headers.Add("Accept","application/json")
+    Invoke-WebRequest -Uri http://localhost:$sideChainAPIPort/api/Node/shutdown -Method Post -ContentType application/json-patch+json -Headers $Headers -Body "true" -ErrorAction SilentlyContinue | Out-Null
+
+    While ( Test-Connection -TargetName 127.0.0.1 -TCPPort $sideChainAPIPort -ErrorAction SilentlyContinue )
+    {
+        Write-Host "Waiting for node to stop..." -ForegroundColor Yellow
+        Start-Sleep 5
+    }
+
+     Write-Host "SUCCESS: Sidechain Node shutdown" -ForegroundColor Green
+     Write-Host ""
+}
+
+function Get-MaxHeight 
+{
+    $Height = @{}
+    $Peers = Invoke-WebRequest -Uri http://localhost:$API/api/ConnectionManager/getpeerinfo | ConvertFrom-Json
+    foreach ( $Peer in $Peers )
+    {
+        if ( $Peer.subver -eq "StratisNode:0.13.0 (70000)" )
+        {
+        }
+            Else
+            {
+                $Height.Add($Peer.id,$Peer.startingheight)
+            }
+    }    
+    
+    $MaxHeight = $Height.Values | Measure-Object -Maximum | Select-Object -ExpandProperty Maximum
+    $MaxHeight
+}       
+
+function Get-LocalHeight 
+{
+    $StatsRequest = Invoke-WebRequest -Uri http://localhost:$API/api/Node/status
+    $Stats = ConvertFrom-Json $StatsRequest
+    $LocalHeight = $Stats.blockStoreHeight
+    $LocalHeight
+}
+
+function Get-LocalIndexerHeight 
+{
+    $IndexStatsRequest = Invoke-WebRequest -Uri http://localhost:$API/api/BlockStore/addressindexertip
+    $IndexStats = ConvertFrom-Json $IndexStatsRequest
+    $LocalIndexHeight = $IndexStats.tipHeight
+    $LocalIndexHeight
+}
+
+function Get-BlockStoreStatus
+{
+    $FeatureStatus = Invoke-WebRequest -Uri http://localhost:$API/api/Node/status | ConvertFrom-Json | Select-Object -ExpandProperty featuresData
+    $BlockStoreStatus = $FeatureStatus | Where-Object { $_.namespace -eq "Stratis.Bitcoin.Features.BlockStore.BlockStoreFeature" }
+    $BlockStoreStatus.state
+}
+
+function Shutdown-Dashboard
+{
+    Write-Host "Shutting down Stratis Masternode Dashboard..." -ForegroundColor Yellow
+    Start-Job -ScriptBlock { Invoke-WebRequest -Uri http://localhost:37000/shutdown } | Out-Null
+        While ( Test-Connection -TargetName 127.0.0.1 -TCPPort 37000 -ErrorAction SilentlyContinue )
+        {
+            Write-Host "Waiting for Stratis Masternode Dashboard to shut down" -ForegroundColor Yellow
+            Start-Sleep 5
+        }
+}
+
+function Check-TimeDifference
+{
+    Write-Host "Checking UTC Time Difference" -ForegroundColor Cyan
+    $timeDif = New-TimeSpan -Start (Get-Date).ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ") -End ( Invoke-WebRequest http://worldtimeapi.org/api/timezone/UTC | ConvertFrom-Json | Select-Object -ExpandProperty utc_datetime ) | Select-Object -ExpandProperty TotalSeconds
+    if ( $timeDif -gt 8 )
```

Comment:
Ideally we should ensure that there is no time difference but looking at what results I'm getting with my system clock being automatically synchronised against time server I would suggest a maximum of 2 second time drift.

Also we should take three samples and get a median as this will produce more unified results.

```
function Check-TimeDifference
{
    Write-Host "Checking UTC Time Difference" -ForegroundColor Cyan
    $timeDifSamples = @([int16]::MaxValue,[int16]::MaxValue,[int16]::MaxValue)
    $timeDifSamples[0] = New-TimeSpan -Start (Get-Date).ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ") -End ( Invoke-WebRequest http://worldtimeapi.org/api/timezone/UTC | ConvertFrom-Json | Select-Object -ExpandProperty utc_datetime ) | Select-Object -ExpandProperty TotalSeconds
    $timeDifSamples[1] = New-TimeSpan -Start (Get-Date).ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ") -End ( Invoke-WebRequest http://worldtimeapi.org/api/timezone/UTC | ConvertFrom-Json | Select-Object -ExpandProperty utc_datetime ) | Select-Object -ExpandProperty TotalSeconds
    $timeDifSamples[2] = New-TimeSpan -Start (Get-Date).ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ") -End ( Invoke-WebRequest http://worldtimeapi.org/api/timezone/UTC | ConvertFrom-Json | Select-Object -ExpandProperty utc_datetime ) | Select-Object -ExpandProperty TotalSeconds

    $timeDif = Get-Median -numberSeries $timeDifSamples

    if ( $timeDif -gt 2 )
    {
        Write-Host "ERROR: System Time is not accurate. Currently $timeDif seconds diffence with actual time! Correct Time & Date and restart" -ForegroundColor Red
        Start-Sleep 30
        Exit
    }
        Else
        {
            Write-Host "SUCCESS: Time difference is $timeDif seconds" -ForegroundColor Green
            Write-Host ""
        }
}

function Get-Median($numberSeries)
{
    $sortedNumbers = @($numberSeries | Sort-Object)
    if ($numberSeries.Count % 2) {
        # Odd, pick the middle
        $sortedNumbers[(($sortedNumbers.Count - 1) / 2)]
    } else {
        # Even, average the middle two
        ($sortedNumbers[($sortedNumbers.Count / 2)] + $sortedNumbers[($sortedNumbers.Count / 2) - 1]) / 2
    }
}
```